samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js

```diff
@@ -27,10 +27,11 @@ class QnAMakerBaseDialog extends QnAMakerDialog {
      * Core logic of QnA Maker dialog.
      * @param {QnAMaker} qnaService A QnAMaker service object.
      */
-    constructor(knowledgebaseId, authkey, host) {
-        var noAnswer = ActivityFactory.fromObject(DefaultNoAnswer);
+    constructor(knowledgebaseId, authkey, host, defaultAnswer) {
+
+        var defaultAnswerActivity = ActivityFactory.fromObject(defaultAnswer?.trim() ? defaultAnswer : DefaultAnswer);
```

Comment:
This pattern differs from the .NET sample implementation, since we've moved away from a C# `const` pattern I think it would be prudent to have a mirror implementation.

Unless the signatures for the QnAMakerDialog differs across languages, is this the case?

Reply:
In javascript .env file is read before constructor is called, whereas in C# config (appsettings.json) is read in constructor.  @stevengum, Please advise how mirror implementation can be done if possible in this case. @gurvsing , can you please take a look here and add if I am missing anything.